### PR TITLE
Support ICU

### PR DESCRIPTION
**Pull request by ry**

### CODE REVIEW

File: src/icu.rs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+// TODO(ry) Map error code to something useful.
+pub fn set_common_data(data: &'static [u8]) -> Result<(), i32> {
+  let mut error_code = 0i32;
+  unsafe {
+    udata_setCommonData_68(data.as_ptr(), &mut error_code as &mut i32);
+  }
+  if error_code == 0 {
+    Ok(())
+  } else {
+    Err(error_code)
```
Comment (piscisaureus) [inline src/icu.rs:44]: Can this actually fail?

File: src/icu.rs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+/// (for example, via opening a converter, constructing a UnicodeString from default-codepage data,
+/// using formatting or collation APIs, etc.), then the default data will be first in the list.
+///
+/// This function has no effect on application (non ICU) data. See udata_setAppData() for similar
+/// functionality for application data.
+// TODO(ry) Map error code to something useful.
+pub fn set_common_data(data: &'static [u8]) -> Result<(), i32> {
+  let mut error_code = 0i32;
+  unsafe {
+    udata_setCommonData_68(data.as_ptr(), &mut error_code as &mut i32);
```
Comment (piscisaureus) [inline src/icu.rs:39]: `as &mut i32` is unnecessary.

File: tests/test_api.rs [hunk 3]
```diff
@@ -4715,3 +4715,46 @@ fn prepare_stack_trace_callback() {
...
+  assert!(v8::icu::set_common_data(ICU_DATA).is_ok());
+
+  let _setup_guard = setup();
+  let isolate = &mut v8::Isolate::new(Default::default());
+  {
+    let scope = &mut v8::HandleScope::new(isolate);
+    let context = v8::Context::new(scope);
+    let scope = &mut v8::ContextScope::new(scope, context);
+    let source = r#"
+        (new Date(Date.UTC(2020, 5, 26, 7, 0, 0))).toLocaleString("de-DE", {weekday: 'long', year: 'numeric', month: 'long', day: 'numeric' });
```
Comment (piscisaureus) [inline tests/test_api.rs:4733]: strange formatting

File: tests/test_api.rs [hunk 4]
```diff
@@ -4715,3 +4715,46 @@ fn prepare_stack_trace_callback() {
...
+  assert!(v8::icu::set_common_data(ICU_DATA).is_ok());
+
+  let _setup_guard = setup();
+  let isolate = &mut v8::Isolate::new(Default::default());
+  {
+    let scope = &mut v8::HandleScope::new(isolate);
+    let context = v8::Context::new(scope);
+    let scope = &mut v8::ContextScope::new(scope, context);
+    let source = r#"
+        new Intl.NumberFormat('ja-JP',{ style: 'currency',currency: 'JPY'}).format(1230000);
```
Comment (piscisaureus) [inline tests/test_api.rs:4753]: strange formatting

**Review by piscisaureus — APPROVED** [review]

File: src/icu.rs (hunk 1 above)
Comment (ry) [reply src/icu.rs:44]: yes, added a test

**Pull request merged**